include/ddc/detail/type_seq.hpp

```diff
@@ -101,6 +101,70 @@ struct TypeSeqMerge<TypeSeq<TagsA...>, TypeSeq<HeadTagsB, TailTagsB...>, TypeSeq
 {
 };
 
+/// A is replaced by element of C at same position than the first element of B equal to A.
+/// Remark : It may not be usefull in its own, it is an helper for TypeSeqReplace
+template <class TagA, class TagSeqB, class TagSeqC>
+struct TypeSeqReplaceSingle;
+
+template <class TagA>
+struct TypeSeqReplaceSingle<TagA, TypeSeq<>, TypeSeq<>>
+{
+    using type = TagA;
+};
+
+template <class TagA, class... TagsB>
+struct TypeSeqReplaceSingle<TagA, TypeSeq<TagsB...>, TypeSeq<>>
+{
+    using type = TagA;
+};
+
+template <class TagA, class... TagsC>
+struct TypeSeqReplaceSingle<TagA, TypeSeq<>, TypeSeq<TagsC...>>
+{
+    using type = TagA;
+};
+
```

Comment:
```suggestion
```

I think it can work without these two specializations. Am I correct that we only want to cover the case where `B` and `C` have the same size ?